File: backend/src/main/java/com/festago/school/dto/SchoolResponse.java
package com.festago.school.dto;

import com.festago.school.domain.School;

/**
 * @deprecated API 버저닝이 적용되면 해당 클래스 삭제
 */
@Deprecated(forRemoval = true)
public record SchoolResponse(
    Long id,
    String domain,
    String name) {

    public static SchoolResponse from(School school) {
        return new SchoolResponse(
            school.getId(),
            school.getDomain(),
            school.getName()
        );
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/SchoolResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.School
import kotlinx.serialization.Serializable

@Serializable
data class SchoolResponse(
    val id: Int,
    val domain: String,
    val name: String
) {
    fun toDomain(): School = School(
        id = id.toLong(),
        domain = domain,
        name = name
    )
}
